src/utils/components/DiskModal/BlankDiskModal.tsx

```diff
@@ -0,0 +1,95 @@
+
+import { PendingChangesAlert } from '../PendingChanges/PendingChangesAlert/PendingChangesAlert';
+import TabModal from '../TabModal/TabModal';
+
+import AdvancedSettings from './components/AdvancedSettings/AdvancedSettings';
+import BootSourceCheckbox from './components/BootSourceCheckbox/BootSourceCheckbox';
+import DiskInterfaceSelect from './components/DiskInterfaceSelect/DiskInterfaceSelect';
+import DiskNameInput from './components/DiskNameInput/DiskNameInput';
+import DiskSizeInput from './components/DiskSizeInput/DiskSizeInput';
+import DiskTypeSelect from './components/DiskTypeSelect/DiskTypeSelect';
+import StorageClassAndPreallocation from './components/StorageClassAndPreallocation/StorageClassAndPreallocation';
+import { getDefaultCreateValues, getDefaultEditValues } from './utils/form';
+import { checkDifferentStorageClassFromBootPVC, diskModalTitle } from './utils/helpers';
+import { submit } from './utils/submit';
+import { SourceTypes, V1DiskFormState, V1DiskModalProps } from './utils/types';
+
+const BlankDiskModal: FC<V1DiskModalProps> = ({
+  editDiskName,
+  isCreated,
+  isOpen,
+  onClose,
+  onSubmit,
+  vm,
+}) => {
+  const { t } = useKubevirtTranslation();
+  const isVMRunning = isRunning(vm);
+
+  const isEditDisk = !isEmpty(editDiskName);
+
+  const initialBootSource = getBootDisk(vm)?.name;
+
+  const isInitialBootDisk = initialBootSource === editDiskName;
+
+  const methods = useForm<V1DiskFormState>({
+    defaultValues: isEmpty(editDiskName)
+      ? getDefaultCreateValues(vm, SourceTypes.BLANK)
+      : getDefaultEditValues(vm, editDiskName),
+    mode: 'all',
+  });
+
+  const {
+    formState: { isSubmitting, isValid },
+    handleSubmit,
+  } = methods;
+
+  return (
+    <FormProvider {...methods}>
+      <TabModal
+        closeOnSubmit={isValid}
+        headerText={diskModalTitle(isEditDisk, isVMRunning)}
+        isLoading={isSubmitting}
+        isOpen={isOpen}
+        onClose={onClose}
+        onSubmit={() => handleSubmit(async (data) => submit(data, vm, editDiskName, onSubmit))()}
+      >
+        {isVMRunning && (
+          <PendingChangesAlert title={t(' Adding hot plugged disk')}>
+            {t(
+              'Additional disks types and interfaces are available when the VirtualMachine is stopped.',
+            )}
+          </PendingChangesAlert>
+        )}
+        <Form>
+          <BootSourceCheckbox
+            initialBootDiskName={initialBootSource}
+            isDisabled={isVMRunning}
+            isInitialBootDisk={isInitialBootDisk}
+          />
+          <DiskNameInput />
+          <DiskSizeInput isCreated={isCreated} />
+          <DiskTypeSelect isVMRunning={isVMRunning} />
+          <DiskInterfaceSelect isVMRunning={isVMRunning} />
+          {!isCreated && (
+            <StorageClassAndPreallocation
+              checkSC={(selectedStorageClass) =>
+                checkDifferentStorageClassFromBootPVC(vm, selectedStorageClass)
+              }
```

Comment:
u are passing this func 4 times, why not consuming it directly at StorageClassAndPreallocation ? just pass VM , selectedSC